tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,571 @@
+      booted_checksum: "{{ head_csum }}"
+
+    - name: Cleanup
+      command: rpm-ostree cleanup -rpmb
+
+#####################################################################################
+#
+# rpm-ostree initramfs
+#   - test client side initramfs
+#
+#####################################################################################
+
+- name: rpm-ostree - upgrade and rebase + install and uninstall
+  hosts: all
+  become: yes
+
+  tags:
+    - initramfs
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - name: Create initramfs file
+      file:
+        path: /etc/rpmostree-initramfs-testing-file
+        state: touch
+
+    - name: Enable initramfs
+      command: rpm-ostree initramfs --enable --arg="-I" --arg="/etc/rpmostree-initramfs-testing-file"
+
+    - include: ../../common/ans_reboot.yml
+
+    - name: Get json output from rpm-ostree status
+      command: rpm-ostree status --json
+      register: ros_status
+
+    - name: Convert to json
+      set_fact:
+         ross_json: "{{ ros_status.stdout | from_json }}"
+
+    - name: Fail if initramfs deployment is not booted
+      fail:
+        msg: Initramfs is not booted
+      when: not ross_json['deployments'][0]['booted']
+
+    - name: Fail if initramfs is not enabled
+      fail:
+        msg: Initramfs is not enabled
+      when: not ross_json['deployments'][0]['regenerate-initramfs']
+
+    - name: Fail if initramfs does not have two arguments
+      fail:
+        msg: "Incorrect number of arguments.  Expected: 2.  Actual: {{ ross_json['deployments'][0]['initramfs-args'] | length }}"
+      when: "{{ ross_json['deployments'][0]['initramfs-args'] | length }} != 2"
+
+    - name: Fail if first initramfs argument is not set
+      fail:
+        msg: "Initramfs args are incorrect.  Expected -I.  Actual {{ ross_json['deployments'][0]['initramfs-args'][0] }}"
+      when: "'-I' not in ross_json['deployments'][0]['initramfs-args'][0]"
+
+    - name: Fail if second initramfs argument is not set
+      fail:
+        msg: "Initramfs args are incorrect.  Expected /etc/rpmostree-initramfs-testing-file.  Actual {{ ross_json['deployments'][0]['initramfs-args'][1] }}"
+      when: "'/etc/rpmostree-initramfs-testing-file' not in ross_json['deployments'][0]['initramfs-args'][1]"
+
+    - name: set osname
+      set_fact:
+        osname: "{{ ross_json['deployments'][0]['osname'] }}"
+
+    - name: Get bootloader entry
+      shell: grep ^initrd /boot/loader/entries/ostree-{{ osname }}-0.conf | sed -e 's,initrd ,/boot/,'
+      register: initramfs
+
+    - name: Check initramfs file
+      command: test -n {{ initramfs.stdout }}
+
+    - name: Get contents of initramfs
+      command: lsinitrd {{ initramfs.stdout }}
+      register: lsinitrd
+
+    - name: Fail if contents of initramfs is incorrect
+      fail:
+        msg: "Expected /etc/rpmostree-initramfs-testing-file in initramfs"
+      when: "'/etc/rpmostree-initramfs-testing-file' not in lsinitrd.stdout"
```

Comment:
@jlebon when I run `lsinitrd $initrd -f /my/file` it produces no file.  Am I running the command right?

```
[root@fedora ~]# ll
total 16
-rw-------. 1 root root 5538 Apr 19 13:13 anaconda-ks.cfg
-rw-------. 1 root root 5196 Apr 19 13:13 original-ks.cfg
[root@fedora ~]# lsinitrd /boot/ostree/fedora-atomic-58da926b4642c6974a8b357b074f3ad3cf168163fb28c42cf3a010eda0ab4b84/initramfs-4.11.0-0.rc6.git0.1.fc26.x86_64.img -f ./myfile.txt
[root@fedora ~]# ll
total 16
-rw-------. 1 root root 5538 Apr 19 13:13 anaconda-ks.cfg
-rw-------. 1 root root 5196 Apr 19 13:13 original-ks.cfg
[root@fedora ~]# lsinitrd /boot/ostree/fedora-atomic-58da926b4642c6974a8b357b074f3ad3cf168163fb28c42cf3a010eda0ab4b84/initramfs-4.11.0-0.rc6.git0.1.fc26.x86_64.img 
Image: /boot/ostree/fedora-atomic-58da926b4642c6974a8b357b074f3ad3cf168163fb28c42cf3a010eda0ab4b84/initramfs-4.11.0-0.rc6.git0.1.fc26.x86_64.img: 49M
========================================================================
Early CPIO image
========================================================================
drwxr-xr-x   3 root     root            0 Jan  1  1970 .
-rw-r--r--   1 root     root            2 Jan  1  1970 early_cpio
drwxr-xr-x   3 root     root            0 Jan  1  1970 kernel
drwxr-xr-x   3 root     root            0 Jan  1  1970 kernel/x86
drwxr-xr-x   2 root     root            0 Jan  1  1970 kernel/x86/microcode
-rw-r--r--   1 root     root        24070 Jan  1  1970 kernel/x86/microcode/AuthenticAMD.bin
========================================================================
Version: dracut-044-178.fc26

Arguments: --reproducible --gzip -v --add 'ostree' --tmpdir '/tmp' -f --no-hostonly --add 'iscsi' --kver '4.11.0-0.rc6.git0.1.fc26.x86_64' --reproducible --gzip -v --add 'ostree' --tmpdir '/tmp' -f -I '/etc/rpmostree-initramfs-testing-file' --kver '4.11.0-0.rc6.git0.1.fc26.x86_64'
..snip...
```